File: web_1/train.py
import chainer
import numpy as np
from chainer import training
from chainer.training import extensions
import random
import collections
from data import NData
from model import Model

def main(f_on, f_off):
	data = NData.load_data(f_on, f_off)
	random.seed(1)
	np.random.seed(1)
	random.shuffle(data)
	dataset = NData.make_dataset(data)
	epoch = 200
	batchsize = 100
	units = len(data[0][0])
	m = Model(units)
	model = m.get_model()
#	gpu = -1
	
	optimizer = chainer.optimizers.Adam()
	optimizer.setup(model)
	test_ratio = 0.05
	nt = int(len(data)*test_ratio)
	test = dataset[:nt]
	train = dataset[nt:]
	train_iter = chainer.iterators.SerialIterator(train, batchsize)
	test_iter = chainer.iterators.SerialIterator(test, batchsize, repeat=False, shuffle=False)
	updater = training.StandardUpdater(train_iter, optimizer)
	trainer = training.Trainer(updater, (epoch, 'epoch'), out='result')
	trainer.extend(extensions.Evaluator(test_iter, model))
	trainer.extend(extensions.dump_graph('main/loss'))
	trainer.extend(extensions.snapshot(), trigger=(epoch, 'epoch'))
	trainer.extend(extensions.LogReport())
	trainer.extend(extensions.PrintReport(
				['epoch', 'main/loss', 'validation/main/loss',
				'main/accuracy', 'validation/main/accuracy']))
	trainer.extend(extensions.ProgressBar())
	
	trainer.run()
	m.save('test.model')
	
if __name__ == '__main__':	
	main("on.txt", "off.txt")

File: web_1/train.rb
require 'pycall/import'
include PyCall::Import

pyimport 'chainer'
pyimport 'numpy', as: :np
pyfrom 'chainer', import: :training
pyfrom 'chainer.training', import: :extensions
pyimport 'random'
pyimport 'collections'
PyCall.sys.path.append(__dir__)
pyfrom 'data', import: :NData
pyfrom 'model', import: :Model


	data = NData.load_data("on.txt", "off.txt")
	random.seed(1)
	np.random.seed(1)
	random.shuffle(data)
	dataset = NData.make_dataset(data)
	epoch = 200
	batchsize = 100
	units = data[0][0].to_a.size
	m = Model.(units)
	model = m.get_model()
	
	optimizer = chainer.optimizers.Adam().new
	optimizer.setup(model)
	test_ratio = 0.05
	nt = (data.to_a.size*test_ratio).to_i
	test = dataset[0...nt]
	train = dataset[nt..-1]
	train_iter = chainer.iterators.SerialIterator.(train, batchsize)
	test_iter = chainer.iterators.SerialIterator.(test, batchsize, repeat=false, shuffle=false)
	updater = training.StandardUpdater.(train_iter, optimizer)
	trainer = training.Trainer.(updater, [epoch, 'epoch'], out='result')
#	trainer.append.(extensions.Evaluator(test_iter, model))
#	trainer.append(extensions.dump_graph('main/loss'))
#	trainer.append(extensions.snapshot(), trigger=[epoch, 'epoch'])
#	trainer.extend(extensions.LogReport())
#	trainer.extend(extensions.PrintReport(
#				['epoch', 'main/loss', 'validation/main/loss',
#				'main/accuracy', 'validation/main/accuracy']))
#	trainer.extend(extensions.ProgressBar())
	
	trainer.run()
	m.save('test.model')
